src/main/java/core/basesyntax/service/impl/TransactionProcessorImpl.java

```diff
@@ -0,0 +1,21 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.TransactionProcessor;
+import core.basesyntax.strategy.OperationStrategy;
+import core.basesyntax.strategy.OperationStrategyImpl;
+import java.util.List;
+
+public class TransactionProcessorImpl implements TransactionProcessor {
+    private OperationStrategyImpl strategyImplementation = new OperationStrategyImpl();
+
+    @Override
+    public void process(List<FruitTransaction> transactions) {
+
+        for (FruitTransaction transaction : transactions) {
+            OperationStrategy operationStrategy = strategyImplementation
+                    .getOperationStrategy(transaction);
+            operationStrategy.calculate(transaction);
+        }

```

Comment:
```suggestion
        for (FruitTransaction transaction : transactions) {
            strategyImplementation
                .getOperationStrategy(transaction)
                .calculate(transaction);
        }
```
Also, it would be cool to practice with forEach method